temporal/queue/queue_test.go

```
@@ -0,0 +1,286 @@
+package queue
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/TykTechnologies/storage/temporal/internal/testutil"
+	"github.com/TykTechnologies/storage/temporal/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestQueue_Publish(t *testing.T) {
+	connectors := testutil.TestConnectors(t)
+	defer testutil.CloseConnectors(t, connectors)
+
+	testCases := []struct {
+		name       string
+		channel    string
+		message    string
+		wantErr    bool
```

Can we change the `wantErr bool` for `expectedErr error`? Ideally, we need to raise the same errors independent of the driver.